### DISCUSSION TITLE
JK-BMS (JK-B2A8S20P-HC) - Am 3-PIN RS-485 Port fehlt GND; Wie RS-485 verbinden?

### DISCUSSION THREAD

**Discussion by swingstate** [hoylabs/OpenDTU-OnBattery#1318] [General]
Hallo, 

ich habe mir ein JK-B2A8S20P-HC (Mit Heizfunktion) und dazu zwei 24V Heizelemente für die Batterie im Garten gekauft. Soweit, so gut. Allerdings hat der RS-485 Port zwar drei Pins, jedoch führt das beigelegte Kabel nur A und B, aber kein GND nach außen. 

![IMG_4287](https://github.com/user-attachments/assets/89884eb3-f133-4c73-9daa-4f4d358e0767)

Hat jemand von euch vielleicht ein ähnliches BMS im Einsatz und kann mir verraten wie er die GND Verbindung zwischen BMS und ESP gelöst hat. Soweit ich weiß sollte neben A und B auch immer GND verbunden sein? Oder nutzt man be den B2A8S20P Modellen auch den GPS Port, in den der JK RS485 Adapter passt?

Vielen Dank!

Bisher hatte ich ein kleines BD4A17S4P im Einsatz, an dem ich auch den JK RS485 Adapter genutzt habe.

**Comment by SW-Niko** [comment]
RS485 sendet symmetrisch und verträgt normalerweise Differenzspannungen von 12V zwischen Sender und Empfänger.
Es geht damit meistens auch ohne GND Verbindung.
Wenn du ein Problem hast, dann kannst du ja mal die Spannung zwischen GND BMS und GND ESP32 messen. Oft reicht schon eine hochohmige Masseverbindung 1k-10k zwischen den Massen damit die RS485 funktioniert.
Die Kombination von RS485 + galvanischer Trennung erfordert unter Umständen GND und VCC.

**Comment by schlimmchen** [comment]
Wie @SW-Niko schon schrieb: Du kannst im Falle von RS485 (in aller Regel) drauf verzichten.

**Comment by swingstate** [comment]
Werde es testen und berichten. Bei dem -HC BMS ist wohl ein CAN (Transceiver?) onboard. Werds vielleicht mal aufschrauben. Über den GPS Port geht wohl weiterhin RS485 wie bei vielen anderen JK's.

**Comment by hoschiking** [comment]
In diesem Falle kannst du nicht nur auf GND verzichten, du solltest es sogar. RS485 ist eine symmetrische Schnittstelle und benötigt für die Funktion kein GND. Wenn du dies machst, verzichtest du auf die galvanische Trennund des GND mit allen Nachteilen.
Man kann GND auf jeder Seite der Schnittstelle GND verbinden, das würde nichts ausmachen, aber nicht zwischen Sender und Empfänger, da genügt eine Simple 2Draht-Leitung, am besten verdrillt

**Reply by SW-Niko** [reply]
> In diesem Falle kannst du nicht nur auf GND verzichten, du solltest es sogar. RS485 ist eine symmetrische Schnittstelle und benötigt für die Funktion kein GND

Im Prinzip ist die Aussage richtig. Die symmetrische Übertragung hat aber auch Grenzen. Bei RS485 liegt sie bei +12V bis -7V, wenn ich das noch richtig im Kopf habe. Mit anderen Worten ... wenn eine Seite floating ist dann geht es meistens nicht.
Symmetrisch und galvanisch getrennt sind streng genommen verschiedene Dinge.

**Reply by hoschiking** [reply]
Natürlich sind es 2 verschiedene Dinge, in diesem Kontext geht es nur darum, dass gnd nicht verbunden wird, das ist damit gegeben. 
Viele BMS trennen den Minuspol (JK ebenfalls), ohne Trennung fließt dann im ungünstigen Fall der Batteriestrom über die Datenleitung....... Daher ist es wichtig

**Comment by swingstate** [comment]
Ich verwende nun den GPS Port zusammen mit dem JK RS485 Adapter an dem von mir designten Board. Das läuft bestens, so wie bei dem kleineren BMS zuvor. Die Beschriftung am BMS ist mit "GPS" etwas irreführend. 
Die Heizfunktion läuft auch und ist recht simpel zu installieren. Ich habe 8 Zellen mit 3x 24V Heizfolien eingepackt. Die Folien sind mittels wärmeleitendem Doppenklebeband an den Zellen befestigt. Drumrum habe ich dann noch Glasfaserklebeband gewickelt. Die Folien ziehen bei 26,6V knapp 1,8A aus der Batterie. Das BMS erlaubt wohl bis 3A (Habe ich gelesen, getestet habe ich bis 2,4A). Nach 10 min sind die Folien ca. 30 - 35° warm. Das Ganze dann in einer Armaflex-isolierten Eurobox.

**Reply by SW-Niko** [reply]
Sehr gut. 👍 
Ich habe schon nach Unterlagen für den JK RS485 Adapter gesucht ... aber nichts gefunden. Macht der eine galvanische Trennung?

**Reply by schlimmchen** [reply]
Scheint so. Ich kann keinen Durchgang messen vom GND auf der RS485 Seite zum GND der Seite mit dem GPS-Stecker.
